### PR TITLE
Convert status codes to `HTTPStatus` in `synapse.rest.admin`

### PR DESCRIPTION
**Pull request by dklimpel**
No functional changes
Only code cleanup

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

File: synapse/rest/admin/_base.py [hunk 1]
```diff
@@ -62,4 +63,4 @@ async def assert_user_is_admin(auth: Auth, user_id: UserID) -> None:
     """
     is_admin = await auth.is_server_admin(user_id)
     if not is_admin:
-        raise AuthError(403, "You are not a server admin")
+        raise AuthError(HTTPStatus.UNAUTHORIZED, "You are not a server admin")
```
Comment (squahtx) [inline synapse/rest/admin/_base.py:66]:
```suggestion
        raise AuthError(HTTPStatus.FORBIDDEN, "You are not a server admin")
```
Was the change to 401 intentional?

File: synapse/rest/admin/media.py [hunk 2]
```diff
@@ -211,11 +212,11 @@ async def on_GET(
         requester = await self.auth.get_user_by_req(request)
         is_admin = await self.auth.is_server_admin(requester.user)
         if not is_admin:
-            raise AuthError(403, "You are not a server admin")
+            raise AuthError(HTTPStatus.UNAUTHORIZED, "You are not a server admin")
```
Comment (squahtx) [inline synapse/rest/admin/media.py:215]:
```suggestion
            raise AuthError(HTTPStatus.FORBIDDEN, "You are not a server admin")
```

File: synapse/rest/admin/users.py [hunk 3]
```diff
@@ -507,7 +531,7 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
         want_mac = want_mac_builder.hexdigest()
 
         if not hmac.compare_digest(want_mac.encode("ascii"), got_mac.encode("ascii")):
-            raise SynapseError(403, "HMAC incorrect")
+            raise SynapseError(HTTPStatus.UNAUTHORIZED, "HMAC incorrect")
```
Comment (squahtx) [inline synapse/rest/admin/users.py:534]:
```suggestion
            raise SynapseError(HTTPStatus.FORBIDDEN, "HMAC incorrect")
```

File: synapse/rest/admin/users.py [hunk 4]
```diff
@@ -712,16 +738,16 @@ async def on_GET(
 
         # To allow all users to get the users list
         # if not is_admin and target_user != auth_user:
-        #     raise AuthError(403, "You are not a server admin")
+        #     raise AuthError(HTTPStatus.UNAUTHORIZED, "You are not a server admin")
```
Comment (squahtx) [inline synapse/rest/admin/users.py:741]:
```suggestion
        #     raise AuthError(HTTPStatus.FORBIDDEN, "You are not a server admin")
```

File: synapse/rest/admin/_base.py (hunk 1 above)
Comment (dklimpel) [reply synapse/rest/admin/_base.py:66]: No.

**Review by squahtx — APPROVED** [review]
Looks good!

**Pull request merged**